src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs

```diff
@@ -3849,6 +3849,141 @@ public void MethodUpdate_LambdaParameterRefnessInBody()
+
+            edits.VerifySemantics(
+                expectedDiagnostics: new[] { Diagnostic(RudeEditKind.Update, "L", CSharpFeaturesResources.local_function) });
+        }
+
+        [Fact]
+        public void MethodUpdate_LocalFunctionReturnType_RemoveAttribute()
+        {
+            var src1 = "class C { void M() { [return: A]int L() { return 1; } } }";
+            var src2 = "class C { void M() { int L() { return 1; } } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [void M() { [return: A]int L() { return 1; } }]@10 -> [void M() { int L() { return 1; } }]@10");
+
+            edits.VerifySemantics(
+                expectedDiagnostics: new[] { Diagnostic(RudeEditKind.Update, "L", CSharpFeaturesResources.local_function) });
+        }
+
+        [Fact]
+        public void MethodUpdate_LocalFunctionReturnType_ReorderAttribute()
+        {
+            var src1 = "class C { void M() { [return: A, B]int L() { return 1; } } }";
+            var src2 = "class C { void M() { [return: B, A]int L() { return 1; } } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [void M() { [return: A, B]int L() { return 1; } }]@10 -> [void M() { [return: B, A]int L() { return 1; } }]@10");
+
+            edits.VerifySemantics(
+                expectedDiagnostics: new[] { Diagnostic(RudeEditKind.Update, "L", CSharpFeaturesResources.local_function) });
+        }
+
+        [Fact]
+        public void MethodUpdate_LocalFunctionParameter_AddAttribute()
+        {
+            var src1 = "class C { void M() { void L(int i) { } } }";
+            var src2 = "class C { void M() { void L([A]int i) { } } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [void M() { void L(int i) { } }]@10 -> [void M() { void L([A]int i) { } }]@10");
+
+            edits.VerifySemantics(
+                expectedDiagnostics: new[] { Diagnostic(RudeEditKind.Update, "L", CSharpFeaturesResources.local_function) });
+        }
+
+        [Fact]
+        public void MethodUpdate_LocalFunctionParameter_RemoveAttribute()
+        {
+            var src1 = "class C { void M() { void L([A]int i) { } } }";
+            var src2 = "class C { void M() { void L(int i) { } } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifyEdits(
+                "Update [void M() { void L([A]int i) { } }]@10 -> [void M() { void L(int i) { } }]@10");
+
+            edits.VerifySemantics(
+                expectedDiagnostics: new[] { Diagnostic(RudeEditKind.Update, "L", CSharpFeaturesResources.local_function) });
+        }
+
+        [Fact]
+        public void MethodUpdate_LocalFunctionParameter_ReorderAttribute()
+        {
+            var src1 = "class C { void M() { void L([A, B]int i) { } } }";
```

Comment:
>A, B] [](start = 53, length = 5)

Let's also add a test that changing `[A][B]` to `[A,B]` and vice versa is allowed.

Reply:
Changing to syntax means this _won't_ be allowed, but I'll add the test anyway so it won't be forgotten in future if attribute edits are allowed in some fashion.